src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1500,10 +1500,11 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
         storage_account_name, sas_token, readonly_sas_token = utils.setup_validate_storage_account(cmd.cli_ctx, storage_account, sas_token, resource_group_name)
         if storage_account_name:  # When validated the storage account
             utils.try_upload_log_file(cluster_name, storage_account_name, sas_token, troubleshoot_log_path)
+            utils.collect_periscope_logs(resource_group_name, cluster_name, storage_account_name, sas_token, readonly_sas_token, kube_context, kube_config)
```

Comment:
These two steps are in both if and else. Lets keep it outside to avoid duplication.

Reply:
Sure. I also noticed that the line after,             `utils.try_archive_log_file(troubleshoot_log_path, output_file)` is also in if and else. But I can help put that outside of the if and else too.